### PR TITLE
Add zsh completion function

### PR DESCRIPTION
**Pull request by bobsaintcool**
Did experiment with ZSH completion function. This one is not bullet proof but may help or get other started!

### CODE REVIEW

**Comment by emersion** [comment]
Can you add it to Meson? (See how sway does it if you need inspiration)

**Comment by bobsaintcool** [comment]
Will do, however allow me few day for that.

**Comment by bobsaintcool** [comment]
Sorry for the delay I need a bit more time for getting into Meson. As its my first work with this don't hesitate if you would like some change.

I've also added a completion system for the mako command.

File: meson.build [hunk 1]
```diff
@@ -107,3 +112,16 @@ if scdoc.found()
 		)
 	endforeach
 endif
+
+if get_option('zsh-completions')
+	zsh_files = files(
+		'completions/zsh/_makoctl',
+                'completions/zsh/_mako'
```
Comment (foxcpp) [inline meson.build]: Mixed indent here.

Comment (bobsaintcool) [reply meson.build]: I've should have fixed this issue in 05758b6298f6c69028fbb961b44d7a01ccdb8fd4 by using editorconfig, thx.

File: meson.build [hunk 2]
```diff
@@ -107,3 +112,16 @@ if scdoc.found()
...
+
+if get_option('zsh-completions')
+	zsh_files = files(
+		'completions/zsh/_makoctl',
+		'completions/zsh/_mako'
+	)
+	zsh_install_dir = datadir + '/zsh/site-functions'
+
+	install_data(zsh_files,
+			install_dir: zsh_install_dir,
```
Comment (emersion) [inline meson.build]: Extra indent here

Comment (bobsaintcool) [reply meson.build]: Should 3b6603f4dd0528d5fbf7a8c5844c26edb4d4ff52 fix this

**Comment by emersion** [comment]
Can you squash everything into a single commit?

**Comment by emersion** [comment]
It seems like there are 3 commits now, including one unrelated commit.

You'll need to make sure you have this repo as a remote

```
git remote add upstream git@github.com:emersion/mako.git
```

Then rebase your PR

```
git fetch upstream
git rebase upstream/master
```

**Comment by bobsaintcool** [comment]
I did an extra rebase and I think we are in a better state now, with only one commit for the above changes.

File: meson.build [hunk 3]
```diff
@@ -14,6 +14,11 @@ project(
 add_project_arguments('-Wno-unused-parameter', language: 'c')
 add_project_arguments('-Wno-missing-braces', language: 'c')
 
+datadir = get_option('datadir')
+prefix = get_option('prefix')
+config = configuration_data()
```
Comment (emersion) [inline meson.build]: `config` isn't used anywhere, and can be removed

**Comment by emersion** [comment]
>I did an extra rebase and I think we are in a better state now, with only one commit for the above changes.

Indeed, looks much better.

Pro tip: to change the last commit, use `git commit --amend`.

File: meson.build (hunk 3 above)
Comment (bobsaintcool) [reply meson.build]: addressed in bf442f0159c87de04e18429b3e5da123680a7069

**Comment by bobsaintcool** [comment]
> Pro tip: to change the last commit, use `git commit --amend`.

Will you change something in the squashed commit message?

**Comment by emersion** [comment]
Thanks!

>Will you change something in the squashed commit message?

For small fixes, probably not

**Pull request merged**